src/Symfony/Component/Validator/Constraints/BicValidator.php

```diff
@@ -81,5 +92,43 @@ public function validate($value, Constraint $constraint)
 
             return;
         }
+
+        // check against an IBAN
+        if ($path = $constraint->ibanPropertyPath) {
+            if (null === $object = $this->context->getObject()) {
+                return;
+            }
+
+            try {
+                $iban = $this->getPropertyAccessor()->getValue($object, $path);
+            } catch (NoSuchPropertyException $e) {
+                throw new ConstraintDefinitionException(sprintf('Invalid property path "%s" provided to "%s" constraint: %s', $path, \get_class($constraint), $e->getMessage()), 0, $e);
+            }
+        } else {
+            $iban = $constraint->iban;
+        }
+        if ($iban) {
```

Comment:
if (!$iban) => return early?

Reply:
@nicolas-grekas no because the logic of validators is to return `null` on the first violation.

If tomorrow we add some more checks after this new code then we don't want to return early in this check.

Reply:
We shouldn't base current code on hypothetical future-based arguments. Here it would make the code clearer *now*.